guide/payments/contacts.md

```diff
@@ -0,0 +1,231 @@
+# Contacts
+
+<!--
+
+
+-->
+
+Whether we’re sending emails, physical mail, or following someone on social media, we primarily think in terms of names and faces, and not the respective address or user ID. Invoice IDs and other identifiers in bitcoin and Lightning are highly unintuitive, and abstracting them via a contact list can create a much smoother user experience. However, there are many [payment request formats]({{ '/guide/payments/request/payment-request-formats/' | relative_url }}), each with unique properties, requiring unique design solutions.
+
+### The basics
+
+<div class="center" markdown="1">
+
+{% include image.html
+   image = "/assets/images/guide/payments/contacts/contact.png"
+   retina = "/assets/images/guide/payments/contacts/contact@2x.png"
+   alt-text = ""
+   width = 250
+   height = 541
+   layout = "float-right-desktop -background -shadow"
+%}
+
+On the surface, a contact entry represents a person, business or other entity that the user sends payment to, or receives payments from.
+
+- A name
+- An image (profile, avatar, company logo…)
+- A list of identifiers
+- Options
+- Prior interactions
+- Organizational features like tags & favorites
+
+Typical user flows involving contacts are:
+
+- Choosing a contact to send bitcoin to
+- Assigning a contact to a generated payment request
+- Browsing the wallet payment history
+- Organizing contacts (merging duplicates, deletion, etc)
+
+</div>
+
+Let’s go over each of these.
+
+### Contact names & images
+
+{% include image-gallery.html pages = page.imagesNames %}
+
+Various payment requests may include labels and descriptions that may be interpreted as contact names. Similarly, Lightning nodes may have aliases assigned to serve as more human-friendly identifiers. Since all of these are optional, a contact system cannot fully rely on automatically detecting and organizing contacts for the user.
+
+A second aspect of this is that these names can be easily spoofed, there is no guarantee that they are accurate. So wallets should parse these identifiers and suggest them to users for their approval.
+Identifiers
+When we think of traditional contacts, we think of phone numbers, emails, and physical addresses. These identifiers are typically static and reliable for long periods of time. This is different in bitcoin, as each payment has its own address or invoice with a unique identifier. There are several approaches to re-usable identifiers (payment request formats), each with its own advantages and disadvantages.
+
+##### On-chain identifiers
+
+<div class="center" markdown="1">
+
+{% include image.html
+   image = "/assets/images/guide/payments/contacts/on-chain-identifiers.png"
+   retina = "/assets/images/guide/payments/contacts/on-chain-identifiers@2x.png"
+   alt-text = ""
+   width = 250
+   height = 116
+   layout = "float-right-desktop -background -shadow"
+%}
+
+There are no good options for re-usable identifiers for on-chain payments, primarily due to privacy concerns.
```

Comment:
PayNym?

Reply:
Works fine with Samourai, what if a product designer interested in creating a privacy focused application reads the guide - they would want to be aware of something like this. We could have a disclaimer saying it's not an official BIP but it could still be used if your user base if some product stacks if it caters to their target audience. We talk about BOLT12 and other non-standards throughout the guide, what's the difference with this? 

> Unanimously Discourage for implementation

Where is this from? Say's its still in draft mode on the bips GitHub repo - https://github.com/bitcoin/bips. Unanimously by who exactly?

Reply:
I don't know but it seems we should find out what the concerns were before promoting its usage in the guide. Samourai's response to you on Twitter wasn't exactly enlightening. We aren't "seeking permission" we're "seeking understanding in order to be responsible."

Reply:
[Reading up](https://github.com/Multibit-Legacy/multibit-hd/issues/818#issuecomment-161673066) on this a bit, here's my understanding:
- Using a payment code requires the sender to create a notification transaction to the receiver, which establishes a log of this relationship (forever)
- Both parties generate a new wallet for this relationship
- The sender can generate addresses to send to
- The receiver generates addresses to observe for incoming transactions

Problems brought up:
- Notification addresses are spam. Since they have to be created between every two unique sender-receiver pair, this could get pretty extensive
- Wallet software needs to manage lots more wallets (dozens?) and observe even more addresses (hundreds, thousands?)

Counterpoint from [SamouraiDev](https://twitter.com/SamouraiDev/status/1345346094692249601?s=20) is that notification transactions are not needed if a good meta data scheme is used. Not sure how that works though. If a wallet is not notified of someone generating addresses and sending transactions, how does it know that they even exist? Should it just scan all transactions forever and match it against a million generated wallets and addresses to see if something is there?

As we've discussed a few times, including the jam session yesterday, how valuable is having a short identifier as a feature? What are the use cases? Are they important enough to warrant this overhead?

Reply:
[One more](https://twitter.com/LukeDashjr/status/1495978585857073154?s=20), somewhat short and cryptic.

Reply:
@GBKS The point of the notification transaction is that the "payment channel" connection is very easily recoverable from the blockchain. It's a tradeoff but it's not necessary. And definitely not a spam. If you don't make notification transaction, you can come up with any other scheme on how the recipient should be notified so they can create a shared secret, derive addresses and listen for incoming transactions.

"unanimously discouraged" stamp of disapproval is irrelevant because BIP47 has been working on-chain on mainnet for years and more wallets are implementing it now.

Reply:
@pajasevi thanks for the details. Which other wallets are implementing this?

In terms of creating a "closed system", the notification transaction seems to be a pretty essential part as it works both for automation/communication and for archival.

Now I wish that "unanimously discouraged" always came with a rationale.

Reply:
Apart from Samourai, Sparrow wallet already implemented it, some guys are working on an implementation for BlueWallet (for bounty) and I believe there might be a third as well but can't recall which one. Also, there are BIP47 open-source libraries written in Java, Javascript and Rust for others to use.

RE: notification tx - wallet MUST always listen for incoming notification TXs but it CAN also have an ability to accept notification via some other method. In this case, it won't be recoverable of course and that should be reflected in the UI.

Reply:
Took a quick look at this. Here's the [BlueWallet bounty](https://github.com/BlueWallet/BlueWallet/issues/2883).

My impression is that people are mixing up BIP 47 and PayNyms. From the bounty issue:
> Having more wallets support this venmo style method for sending money to your friends, would benefit the ecosystem in general.

Aren't PayNyms a centralized solution by Samourai that gets you nice Twitter-like handles ("+christoph"). [Here's a line](https://github.com/sparrowwallet/sparrow/blob/ce6b371206b1609462199ee26488e657e6b4251d/src/main/java/com/sparrowwallet/sparrow/paynym/PayNymService.java#L55) in the Sparrow codebase that uses the PayNym API (`https://paynym.is/api/`). If so, then I don't think it's a good idea to push this as it's completely centralized.

Reply:
> If so, then I don't think it's a good idea to push this as it's completely centralized.

Playing devils advocate here but - I'd put as having the same tradeoffs as Lightning addresses, trusted but still non-custodial. If we go with that logic why do we recommend Lightning addresses? 

If we introduce any content around this in the guide we should just refer to it as payment codes / BIP47 and not PayNym (Samourai's implementation of the aforementioned). We don't have to recommend to use it or not use it but can just have it mentioned. Product developers can decide if they want to use it not.

Reply:
Paynyms are an implementation of BIP47 payment codes. paynym.is is just a directory of payment codes with generated short name and image for more user-friendly usage. A wallet implementing BIP47 properly doesn't have to use this directory and still will be completely interoperable with other implementations.

Reply:
@pajasevi that's my understanding as well. I just don't think that's clear to the people in the Blue Wallet bounty thread.

@Bosch-0 Lightning addresses rely on servers (for the underlying LNURL service) but are not centralized. Your Lighting address and my Lightning address can be hosted on different servers. Not so with +paynym handles.

Let's see how we can get BIP 47 info added. Will try to think through that.